### PR TITLE
Examples: webgl_postprocessing_hbao

### PR DESCRIPTION
**Pull request by Rabbid76**
WebGL post processing example: HABO (Horizon Based Ambient Occlusion)

*This contribution is funded by [Rabbid76](https://github.com/Rabbid76)*

### CODE REVIEW

File: examples/jsm/shaders/HBAOShader.js [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+
+		}`
+
+};
+
+function generateHaboSampleKernelInitializer( samples, rings ) {
+
+	const poissonDisk = generateHaboSamples(
+		samples,
+		rings,
```
[the remark this answers is not in the document]
Comment (Rabbid76) [reply examples/jsm/shaders/HBAOShader.js]: resolved

**Comment by mrdoob** [comment]
Very nice!

**Comment by mrdoob** [comment]
Hmm... Is this based on [N8AO](https://github.com/N8python/n8ao)?

**Comment by Rabbid76** [comment]
> Hmm... Is this based on [N8AO](https://github.com/N8python/n8ao)?

Sorry I didn't answer right away, but I didn't read your comment until now.  
The basic idea is form [realism-effects](https://github.com/0beqz/realism-effects), since the algorithm there has some flaws, I did some more research and went the way of N8AO. I also took a closer look at the shaders https://github.com/nvpro-samples/gl_ssao/blob/master/hbao.frag.glsl and https://github.com/scanberg/hbao/blob/master/resources/shaders/hbao_frag.glsl. At the heart of these algorithms is something like `dot(N, F) * fallOff`.
Would you prefer a different algorithm for the example?

**Comment by N8python** [comment]
You mean cosine weighting the hemisphere? If you are already cosine-weighting your samples, you don't need to cosine-weight the hemisphere.

**Comment by Rabbid76** [comment]
> You mean cosine weighting the hemisphere? If you are already cosine-weighting your samples, you don't need to cosine-weight the hemisphere.

I see. So I should remove `cosineSampleHemisphere` and calculate the samples differently. I can do that.

**Pull request merged**